upstairs/src/client.rs

```diff
@@ -443,21 +443,25 @@ impl DownstairsClient {
         out
     }
 
-    pub(crate) fn replay_job(&mut self, job: &mut DownstairsIO) {
+    pub(crate) fn replay_job(&mut self, job: &mut DownstairsIO) -> bool {
```

Comment:
Add a docstring here indicating the significance of the returned `bool`, e.g.
```rust
/// Ensures that the given job is in the job queue and in `IOState::New`
///
/// Returns `true` if the job was requeued, or `false` if it was already `New`

Reply:
Updated in 906d5b9